bin/proving-service/src/api/mod.rs

```diff
@@ -21,48 +24,118 @@ pub struct RpcListener {
 }
 
 impl RpcListener {
-    pub fn new(listener: TcpListener) -> Self {
-        let api_service = ApiServer::new(ProverRpcApi::default());
+    pub fn new(listener: TcpListener, is_tx_prover: bool, is_batch_prover: bool) -> Self {
+        let prover_rpc_api = ProverRpcApi::new(is_tx_prover, is_batch_prover);
+        let api_service = ApiServer::new(prover_rpc_api);
         Self { listener, api_service }
     }
 }
 
-#[derive(Default)]
 pub struct ProverRpcApi {
-    local_prover: Mutex<LocalTransactionProver>,
+    tx_prover: Option<Mutex<LocalTransactionProver>>,
+    batch_prover: Option<Mutex<LocalBatchProver>>,
 }
```

Comment:
Related to the previous comment: is there a downside to doing something like:
```rust
pub struct ProverRpcApi {
    provers: Mutex<Provers>,
}

struct Provers {
    tx_prover: LocalTransactionProver,
    batch_prover: LocalBatchProver,
}
```
(there may be a better way of doing this too).

Reply:
The only thing that might be affected using this approach is that currently we are capable of proving the two types of proofs at the same time. With this approach we can only generate one proof at a time, no matter the type of the proof. Though it might actually be better using what you suggested, since we have `concurrent` by default.

Reply:
I implemented this new approach 👍🏼